odoo/upgrade_code/18.1-01-rename-class.py

```diff
@@ -0,0 +1,178 @@
+                    "JOIN " in line or
+                    " UNION " in line or
+                    " AND " in line or
+                    " ON " in line or
+                    " OR " in line or
+                    " WHERE " in line or
+                    " GROUP BY " in line or
+                    " SELECT " in line):
+                    continue
+                line = reg_use_class_name.sub(new_class_name, line)
+                line = reg_use_super.sub(' super()', line)
+                lines[no] = line
+            except ValueError:
+                continue
+    return '\n'.join(lines)
+
+
+def remove_name_and_update_inherit(content):
+    lines = content.split('\n')
+    length = len(lines)
+
+    class_model_name = None
+    for index, line in enumerate(lines):
+        if index >= length:
+            continue
+
+        class_name = line.startswith('class ') and ('Model)' in line or '.' in line) and line[6:].split(':')[0].split('(')[0]
+        if not class_name:
+            continue
+        class_model_name = class_name_to_model_name(class_name)
+        if not class_model_name:
+            continue
+
+        model_name = None
+        index_name = -1
+        inherit_model_name = None
+        index_inherit = -1
+        for i in range(index + 1, length):
+            new_line = lines[i]
+            if new_line.startswith('class ') or line.startswith('def '):
+                break
+            new_model_name = attribute_to_model_name(new_line)
+            if new_model_name:
+                if '_name' in new_line:
+                    model_name = new_model_name
+                    index_name = i
+                elif '_inherit' in new_line and '_inherits' not in new_line:
+                    inherit_model_name = new_model_name
+                    index_inherit = i
+
+        # update `_inherit` = 'xxx' into `_inherit` = ['xxx']
+        if index_inherit != -1:
+            inherit_line = lines[index_inherit]
+            if '[' not in inherit_line:
+                # makes the code idempotent
+                attr, value = inherit_line.split('=')
+                lines[index_inherit] = f'{attr.rstrip()} = [{value.strip()}]'
+
+        if model_name:
+            if class_model_name == model_name:
+                if len(lines[index_name].split('=')) > 2:
+                    lines[index_name] = re.sub(r'_name *= *', '', lines[index_name])
+                else:
+                    lines.pop(index_name)
+                length -= 1
+        elif inherit_model_name and class_model_name != inherit_model_name:
+            inherit_line = lines[index_inherit]
+            tab = inherit_line[0:(len(inherit_line) - len(inherit_line.lstrip()))]
+            lines.insert(index_inherit, f'{tab}_name = "{inherit_model_name}"\n')
+            length += 1
+
+    return '\n'.join(lines)
+
+
+def class_with_2_blank_lines(content):
+    return re.sub(r'(?:\s*\n)+?(# .*\n)?(class .*Model\))', r'\n\n\n\1\2', content)
```

Comment:
Just add a comment for the function: Replace multiple blank lines before class definitions into exactly 2 blank lines.